app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,83 @@
+class VendorsController < ApplicationController
+  def index
+    uri = URI(rmapi_path)
+
+    # Transform query params to what the EBSCO RM-API expects
+    uri.query = URI.encode_www_form(
+      search: params[:q],
+      orderby: params[:orderby] || params[:q] ? 'relevance' : 'vendorname',
+      count: params[:count] || 25,
+      offset: params[:offset] || 1
+    )
+
+    # Make the RM-API request and parse the response body
+    response = make_request(uri)
+    data = JSON.parse(response.body)
```

Comment:
this is fine as-is for now (since we should let patterns emerge before adopting them and whatnot), but as we move forward i'd like to see the logic above this line extracted somewhere.  probably not worth its own gem, but how would you feel about a service object?

```ruby
ResourceManagementService.new(path: rmapi_path, params: params).vendors
```

or, thinking about the cases where we're going to have to do multiple RMAPI calls to surface all necessary data in one response, the service object could accept operations: 

```ruby
ResourceManagementService.new(
  path: rmapi_path, 
  operations: [ 
    GetPackage.new(...),  
    GetCustomerResource.new(...),  
  ]
).process
```
We can cross that bridge when we get to it, but my main point is that we should keep our API layer as thin as possible.

Reply:
I agree, as we were submitting this PR we noted how most other controllers are going to be the same, just subbing the model name and RM-API endpoints/query params